baseline/client/admin/admin.js

```diff
@@ -0,0 +1,68 @@
+import "./style.css";
+
+import { getAuth } from "auth/auth.js";
+import Db from "db/db.js";
+
+const experimentSelect = document.getElementById("experiment-select");
+const experimentButton = document.getElementById("experiment-button");
+const experimentDownload = document.getElementById("experiment-download");
+
+const experimentNames = ["daily-stressors", "dass", "demographics", "face-name", "ffmq", "flanker", "mind-in-eyes", "mood-memory", "mood-prediction", "n-back", "panas", "pattern-separation-learning", "pattern-separation-recall", "physical-activity", "spatial-orientation", "task-switching", "verbal-fluency", "verbal-learning-learning", "verbal-learning-recall"];
+function initializeSelect() {
+    experimentNames.forEach(n => {
+        const option = document.createElement("option");
+        option.value = n;
+        option.label = n;
+        experimentSelect.add(option);
+    });
+    const allOption = document.createElement("option");
+    allOption.value = "";
+    allOption.label = "all";
+    experimentSelect.add(allOption);
+    experimentSelect.removeAttribute("disabled");
+}
+
+function initializeButton(db) {
+    experimentButton.addEventListener("click", async () => {
+        prepareDownload();
+        const results = await db.getResultsForExperiment(experimentSelect.value);
+        const filename = `${experimentSelect[experimentSelect.selectedIndex].label}.json`;
+        enableDownload(results, filename);
+    });
+    experimentButton.removeAttribute("disabled");
+}
+
+function disableDownload() {
+    experimentDownload.href = "";
+    experimentDownload.download = "";
+    experimentDownload.textContent = "...";
+    experimentDownload.setAttribute("aria-disabled", "true");
+}
+
+function prepareDownload() {
+    disableDownload();
+    experimentDownload.textContent = "Querying...";
+}
+
+function enableDownload(results, filename) {
+    experimentDownload.href = URL.createObjectURL(new Blob(
```

Comment:
One of your helpful links (thanks for those!) mentions that you should call revokeObjectURL on object urls if you're able to, as they'll hang around otherwise. Given the size of the ones we're creating here, that's important. Maybe call that after they click on the link (if that doesn't mess up the save) or if they change the experiment selection? Or maybe just after a timeout?

Reply:
Actually, never mind - I realized this will go away when we move the query to the server side.